test/e2e/validation_test.go

```diff
@@ -289,13 +288,31 @@ func (tc *testContext) sshSetup() error {
 	return nil
 }
 
-// runSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed on the host specified
-// by the provided IP.
-func (tc *testContext) runSSHJob(name, command, ip string) (string, error) {
+// runPowershellSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed through
+// powershell, on the host specified by the provided IP.
+func (tc *testContext) runPowershellSSHJob(name, command, ip string) (string, error) {
```

Comment:
Why is this change required? Is this because the job could land on a node that has the PowerShell as default? I also suggest leaving the name as `runSSHJob`. The only reason to make it more verbose would be if we were also running command prompt jobs.

Reply:
This function was required for the same reason we need to determine the default shell in the operator code, running `powershell.exe ... -Command "..."` within powershell, was causing parsing issues that I couldn't resolve.

I created a separate function for this to separate the logic of determining the shell and modifying the command based on that, from the logic of running the job and everything that comes with that.